### PR TITLE
Fix signature of JS_BufferIsCompilableUnit

### PR DESCRIPTION
**Pull request by tschneidereit**
r? @larsbergstrom

[Review on Reviewable](https://reviewable.io/reviews/servo/rust-mozjs/192)

### CODE REVIEW

File: src/jsapi.rs [hunk 1]
```diff
@@ -4490,14 +4490,14 @@ extern "C" {
...
-    fn _Z25JS_BufferIsCompilableUnitP9JSContextN2JS6HandleIP8JSObjectEEPKcj(cx:
+    fn _Z25JS_BufferIsCompilableUnitP9JSContextN2JS6HandleIP8JSObjectEEPKcm(cx:
                                                                                 *mut JSContext,
                                                                             obj:
                                                                                 Handle<*mut JSObject>,
                                                                             utf8:
                                                                                 *const i8,
                                                                             length:
-                                                                                u32)
+                                                                                u64)
```
Comment (Manishearth) [inline src/jsapi.rs]: This is `size_t`. Should we be be using `usize`?

Comment (tschneidereit) [inline src/jsapi.rs]: >
> This is size_t. Should we be be using usize?
>
Probably, yes. It seems like bindgen generated wrong code for size_t? I
wonder if that afflicts more sites than just this one.

Comment (Manishearth) [inline src/jsapi.rs]: cc @michaelwu

Comment (Ms2ger) [inline src/jsapi.rs]: Yes and yes: #185

Comment (Manishearth) [inline src/jsapi.rs]: Is `libc::size_t` different from `usize`? It would be nice to use Rust ints whenever possible.

**Comment by michaelwu** [comment]
https://gist.github.com/michaelwu/a9525f07112a8224a540 is the latest output from bindgen. `size_t` is handled correctly now.

Replacing `size_t` with usize sounds good - I'll try that next time, though I think we should stick to `size_t` for now for consistency.

**Comment by michaelwu** [comment]
This should be fixed by the bindings update.

**Comment by michaelwu** [comment]
https://github.com/servo/rust-mozjs/pull/198 will fix this.

**Pull request closed without merging**